ersilia/core/tracking.py

```diff
@@ -261,28 +276,36 @@ def stats(self, result):
         :return: A dictionary containing the stats for each column of the result.
         """
 
-        dat = pd.read_csv(result)
+        data = read_csv(result)
 
         # drop first two columns (key, input)
-        dat = dat.drop(["key", "input"], axis=1)
-
-        # calculate statistics
+        for row in data:
+            row.pop('key', None)
+            row.pop('input', None)
+            
+        # Convert data to a column-oriented format
+        columns = defaultdict(list)
+        for row in data:
+            for key, value in row.items():
+                columns[key].append(float(value))
+
+        # Calculate statistics
         stats = {}
-        for column in dat:
+        for column, values in columns.items():
             column_stats = {}
-            column_stats["mean"] = dat[column].mean()
-            if len(dat[column].mode()) == 1:
-                column_stats["mode"] = dat[column].mode().iloc[0]
-            else:
+            column_stats["mean"] = statistics.mean(values)
+            try:
+                column_stats["mode"] = statistics.mode(values)
+            except statistics.StatisticsError:
                 column_stats["mode"] = None
-            column_stats["min"] = dat[column].min()
-            column_stats["max"] = dat[column].max()
-            column_stats["std"] = dat[column].std()
+            column_stats["min"] = min(values)
+            column_stats["max"] = max(values)
+            column_stats["std"] = statistics.stdev(values) if len(values) > 1 else 0
 
             stats[column] = column_stats
 
         return stats
-
+    
     def get_file_sizes(self, input_df, output_df):
         """
         Calculates the size of the input and output dataframes, as well as the average size of each row.
```

Comment:
Let's not call these dataframes now, since we are simply using the input and output files now.